src/sidebar/services/api-routes.js

```diff
@@ -6,7 +6,7 @@ const { retryPromiseOperation } = require('../util/retry');
  * A service which fetches and caches API route metadata.
  */
 // @ngInject
-function apiRoutes($http, settings) {
+function apiRoutes(settings) {
```

Comment:
Perhaps add docstring about this function returning a (I believe?) Promise that resolves to JSON?

Reply:
Even though this is service is declared with `function` it is effectively a class, which has a `links` method that returns a Promise (see below). Our Angular services are declared with `function` because they were originally created pre-ES6.